crates/hir/src/lib.rs

```diff
@@ -756,3 +757,11 @@ impl Block {
         }
     }
 }
+
+/// An expr binding in HIR
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Binding {
+    pub name: Name,
+    pub mutable: bool,
```

Comment:
**Performance Issue**: The Binding struct includes a mutable field which is a boolean. Frequent checks and modifications to this field can lead to performance issues in a multi-threaded environment due to data races or lock contention. <br> **Fix**: Consider using atomic types for the mutable field or other synchronization mechanisms to ensure thread safety without significant performance overhead. <br> **Code Suggestion**: 
 ```
 -    pub mutable: bool,
 +    pub mutable: AtomicBool,
 ```